src/components/common/needsShutdown.jsx

```diff
@@ -0,0 +1,168 @@
+    const inactiveIface = nicLookupByMAC(vm.inactiveXML.interfaces, iface.mac);
+
+    return inactiveIface && inactiveIface.type !== iface.type;
+}
+
+export function needsShutdownIfaceModel(vm, iface) {
+    const inactiveIface = nicLookupByMAC(vm.inactiveXML.interfaces, iface.mac);
+
+    return inactiveIface && inactiveIface.model !== iface.model;
+}
+
+export function needsShutdownIfaceSource(vm, iface) {
+    const inactiveIface = nicLookupByMAC(vm.inactiveXML.interfaces, iface.mac);
+
+    return inactiveIface &&
+        getIfaceSourceName(inactiveIface) !== getIfaceSourceName(iface);
+}
+
+export function needsShutdownVcpu(vm) {
+    return ((vm.vcpus.count !== vm.inactiveXML.vcpus.count) ||
+            (vm.vcpus.max !== vm.inactiveXML.vcpus.max) ||
+            (vm.cpu.sockets !== vm.inactiveXML.cpu.sockets) ||
+            (vm.cpu.threads !== vm.inactiveXML.cpu.threads) ||
+            (vm.cpu.cores !== vm.inactiveXML.cpu.cores));
+}
+
+export function needsShutdownCpuModel(vm) {
+    /* The live xml shows what host-model expanded to when started
+     * This is important since the expansion varies depending on the host and so needs to be tracked across migration
+     */
+    if (vm.inactiveXML.cpu.mode === 'host-model')
+        return !(vm.cpu.mode == 'host-model' || vm.cpu.model === vm.capabilities.cpuHostModel);
+    if (vm.inactiveXML.cpu.mode === 'host-passthrough')
+        return vm.cpu.mode !== 'host-passthrough';
+    if (vm.inactiveXML.cpu.mode === 'custom')
+        return vm.cpu.mode !== 'custom' || vm.cpu.model !== vm.inactiveXML.cpu.model;
+
+    return false;
+}
+
+export function needsShutdownWatchdog(vm) {
+    return vm.persistent && vm.state === "running" && vm.inactiveXML.watchdog.action !== vm.watchdog.action;
+}
+
+export function getDevicesRequiringShutdown(vm) {
+    if (!vm.persistent)
+        return [];
+
+    const devices = [];
+
+    // DISKS
+    for (const target in vm.disks) {
+        if (needsShutdownDiskAccess(vm, target)) {
+            devices.push(_("Disk"));
+            break;
+        }
+    }
+
+    // INTERFACES
+    for (const iface of vm.interfaces) {
+        if (needsShutdownIfaceType(vm, iface) ||
+            needsShutdownIfaceModel(vm, iface) ||
+            needsShutdownIfaceSource(vm, iface)) {
+            devices.push(_("Network interface"));
+            break;
+        }
+    }
+
+    // VCPU
+    if (needsShutdownVcpu(vm))
+        devices.push(_("vCPUs"));
+
+    // CPU
+    if (needsShutdownCpuModel(vm))
+        devices.push(_("CPU"));
+
+    // Watchdog
+    if (needsShutdownWatchdog(vm))
+        devices.push(_("Watchdog"));
+
+    return devices;
+}
+
+export const NeedsShutdownTooltip = ({ iconId, tooltipId }) => {
+    return (
+        <Tooltip id={tooltipId} content={_("Changes will take effect after shutting down the VM")}>
+            <PendingIcon color="orange" id={iconId} />
+        </Tooltip>
+    );
+};
+
+NeedsShutdownTooltip.propTypes = {
```

Comment:
Honestly we stopped adding proptypes I think but fine by me. We need to sync about it though.

Reply:
I just renamed this file and renamed the component, the proptypes were already there, but sure I can remove them

Reply:
Up to you - I don't mind